cmd/mcp-server/eval/scripts/analysis_helpers.py

```
@@ -0,0 +1,284 @@
+
+
+@lru_cache(maxsize=128)
+def load_annotations(dataset_dir, scenario_id):
+    """Load annotations.yaml for a scenario (cached to avoid re-reading)."""
+    ann_file = Path(dataset_dir) / scenario_id / "annotations.yaml"
+    if not ann_file.exists():
+        return {}
+    try:
+        with open(ann_file) as f:
+            return yaml.safe_load(f) or {}
+    except Exception:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)